Makefile

```diff
@@ -56,6 +51,18 @@ EXTRA_CLEAN = $(GENERATED_OUT)
 
 PGXS := $(shell $(PG_CONFIG) --pgxs)
 
+ifeq ($(OS), Linux)
+  DL_SUFFIX=so
+else ifeq ($(OS), Darwin)
+  ifeq ($(PG_GE16), 0)
+    DL_SUFFIX=dylib
+  else
+    DL_SUFFIX=so
+  endif
```

Comment:
MacOS only uses `.dylib` from PG16+ https://github.com/NixOS/nixpkgs/blob/7d5cd42fece2ae9b065b00c08696b439a864401f/pkgs/servers/sql/postgresql/generic.nix#L114

Reply:
There is also a variable `DLSUFFIX` on postgres' Makefile:
https://github.com/postgres/postgres/blob/REL_17_0/src/Makefile.global.in#L553
I think it might be used by supautils' Makefile.

Reply:
It doesn't get included with `include $(PGXD)` unfortunately

Reply:
AFAIK it is included by `pgxs.mk`:
https://github.com/postgres/postgres/blob/REL_17_0/src/makefiles/pgxs.mk#L87
but maybe it works a bit differently in `supautils`. Sorry for the noise.